modules/dxp/apps/commerce/commerce-shop-by-diagram-web/src/main/java/com/liferay/commerce/shop/by/diagram/admin/web/internal/util/CSDiagramSettingUtil.java

```diff
@@ -22,12 +22,26 @@
 import com.liferay.portal.kernel.log.Log;
 import com.liferay.portal.kernel.log.LogFactoryUtil;
 import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.repository.model.FileVersion;
 
 /**
  * @author Andrea Sbarra
+ * @author Crescenzo Rega
  */
 public class CSDiagramSettingUtil {
 
+	public static FileVersion getFileVersion(CSDiagramSetting csDiagramSetting)
+		throws Exception {
+
+		FileEntry fileEntry = _fetchFileEntry(csDiagramSetting);
+
+		if (fileEntry != null) {
+			return fileEntry.getFileVersion();
+		}
+
+		return null;
+	}
+
 	public static String getImageURL(
```

Comment:
@andrea-ale-sbarra this is the change you asked for?

Reply:
yes I thought that there was a reason for keeping the fetch private so since this was used just for get the file version I suggested this, but I'm open to other solution if better